### PR TITLE
bump-lockfile: de-parallelize

### PR DESCRIPTION
**Pull request by jlebon**
Right now, the job tries to update both `testing-devel` and `next-devel`
in parallel, but this makes the output messy and hard to follow. It also
makes it harder to tell from the job status page which of the two
branches failed (and the BlueOcean UI isn't much help either).

Instead, let's make `bump-lockfile` focus on a single stream, and have a
driver job which simply triggers `bump-lockfile` once per stream so that
they're separate jobs. This mirrors how the main pipeline jobs
themselves are set up.

Patch best viewed with whitespace ignored.

Closes: #331

### CODE REVIEW

File: jobs/bump-lockfile.Jenkinsfile [hunk 1]
```diff
@@ -1,121 +1,124 @@
...
-                shwrap("cosa fetch --strict")
-                shwrap("cp src/config/fedora-coreos-pool.repo{.bak,}")
-            }
-
-            stage("Build") {
-                shwrap("cosa build --strict")
-            }
-
-            // need to run this in the workspace context because `fcosKola()`
-            // archives directly there but archiveArtifacts is dir-sensitive
```
Comment (dustymabe) [inline jobs/bump-lockfile.Jenkinsfile:74]:
```
// need to run this in the workspace context because `fcosKola()`
// archives directly there but archiveArtifacts is dir-sensitive
```

No longer applicable?

File: jobs/bump-lockfile.Jenkinsfile [hunk 2]
```diff
@@ -1,121 +1,124 @@
...
+        shwrap("git -C src/config commit -am 'lockfiles: bump to latest' -m 'Job URL: ${env.BUILD_URL}' -m 'Job definition: https://github.com/coreos/fedora-coreos-pipeline/blob/master/jobs/bump-lockfile.Jenkinsfile'")
+        withCredentials([usernamePassword(credentialsId: botCreds,
+                                          usernameVariable: 'GHUSER',
+                                          passwordVariable: 'GHTOKEN')]) {
+          // should gracefully handle race conditions here
+          sh("git -C src/config push https://\${GHUSER}:\${GHTOKEN}@github.com/${repo} ${branch}")
         }
-    }] }
+    }
+    currentBuild.description = "[${params.STREAM}] ⚡ (pushed)"
```
Comment (dustymabe) [inline jobs/bump-lockfile.Jenkinsfile:119]: ❤️

File: jobs/bump-lockfile.Jenkinsfile [hunk 3]
```diff
@@ -1,121 +1,124 @@
...
+    // do a first fetch where we only fetch metadata; no point in
+    // importing RPMs if nothing actually changed
+    stage("Fetch Metadata") {
+        shwrap("cosa fetch --update-lockfile --dry-run")
+    }
+
+    def newPkgChecksum = shwrapCapture("jq -c .packages src/config/manifest-lock.*.json | sha256sum")
+    if (newPkgChecksum == prevPkgChecksum) {
+        println("No changes")
+        currentBuild.description = "[${params.STREAM}] 💤 (no change)"
```
Comment (dustymabe) [inline jobs/bump-lockfile.Jenkinsfile:52]: ❤️

**Review by dustymabe — APPROVED** [review]
LGTM - one question..

File: jobs/bump-lockfile.Jenkinsfile (hunk 1 above)
Comment (jlebon) [reply jobs/bump-lockfile.Jenkinsfile:74]: Yeah, because now the workdir is directly in the workspace instead of a subdir.

**Comment by jlebon** [comment]
Let's try this out!
